OpenRA.Game/Widgets/RootWidget.cs

```diff
@@ -31,6 +31,14 @@ public override bool HandleKeyPress(KeyInput e)
 					return true;
 				}
 
+				if (hk == Game.Settings.Keys.HideUserInterfaceKey)
+				{
+					foreach (var child in this.Children)
+						child.Visible = child.Visible ^ true;
```

Comment:
Does toggling this widget's visibility not work?

In any case, `child.Visible ^= true`.